flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/schema/TableSourceTable.scala

```diff
@@ -34,16 +34,21 @@ import org.apache.flink.table.sources.{TableSource, TableSourceUtil}
 class TableSourceTable[T](
     val tableSource: TableSource[T],
     val isStreamingMode: Boolean,
-    val statistic: FlinkStatistic)
+    val statistic: FlinkStatistic,
+    val selectedFields: Option[Array[Int]])
```

Comment:
Can be simplified to `val selectedFields: Option[Array[Int]] = None`. Then we don't need to introduce a new constructor.

Reply:
There are Java invokers, so introduce this constructor.

Reply:
Thanks for the tip.